### ISSUE TITLE
SCons always uses relative path even when absolute path is set with LIBPATH

### ISSUE THREAD

**Issue by msbm97** [SCons/scons#3128]
I am compiling a complex project with multiple shared library which will be linked to a code not using scons so I need the registered paths to be absolute. My sconscript has the following code 

common_build_dir = os.getcwd() + os.sep+ 'build' +os.sep() + 'common'

dynamic_lib = env.SharedLibrary(target = 'dynamic_lib',
                                source = [srcs],
                                LIBS = 'common',
                                LIBPATH= common_build_dir,
                                RPATH = common_build_dir
                               )

When I run otool the output is:
./lib/libTdynamic_lib.dylib:
	build/dynamic_lib/libdynamic_lib.dylib (compatibility version 0.0.0, current version 0.0.0)
	build/common/libcommon.dylib (compatibility version 0.0.0, current version 0.0.0)

and I can see from the build commands that scons is indeed omitting the complete path and only referencing from relative to wheres construct is defined.

PS: I tried joins the scons mailing list but have not received an approval email

**Comment by bdbaddog** [comment]
which version of SCons, python, platform?
Also can you paste the relevant part of your build log?

**Comment by bdbaddog** [comment]
check your spam folder for approval email.

**Comment by msbm97** [comment]
yes that is correct. I am sorry I missed where you were requesting the platform also.

**Comment by msbm97** [comment]
I tried what you suggested and made the following change to my script

newBuildDir = DIR(common_build_dir)

dynamic_lib = env.SharedLibrary(target = 'dynamic_lib',
                                source = srcs,
                                LIBS = 'common',
                                LIBPATH= newBuildDir,
                                RPATH = newBuildDir

Unfortunately, it didn't seem to make any difference the output of otool -L dynamic_lib.dylib still showed the relative path.

Is there anything else I can do to get scons register the absolute path? It will be a pain if the files always register relative paths because I have 7 projects which have  varying levels of dependency on each other.

I got the confirmation email and I am now in the scons user groups.
